.ci/static-checks.sh

```diff
@@ -537,8 +537,25 @@ check_docs()
 		fi
 
 		# Check the URL, saving it if invalid
-		( curl -sLf -o /dev/null "$url" ||\
-				echo "$url" >> "$invalid_urls") &
+		(
+			local http_statuses
+
+			# A single doc can generate multiple HTTP status
+			# codes, so save them all.
+			http_statuses=$(curl -sIL "$url" | grep -E "^HTTP" | awk '{print $2}' || true)
+
+			local status
+
+			for status in $http_statuses
+			do
+				local ret
+
+				# See https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
+				{ echo "$status" | grep -qE "^(2|3)[0-9][0-9]"; ret=$?; } || true
+
+				[ $ret -ne 0 ] && { echo "$url" >> "$invalid_urls" && break; } || true
+			done
+		) &
```

Comment:
hurrrr, is that multiple parallel writers in the background to a single `$invalid_urls` file - heh, one day I expect you to get inter-spersed writes.... but, given that only happens in an error condition, and is probably rare, it's probably not a blocker - but, does look odd ;-)

Reply:
I haven't checked the source, but `strace` suggests this is safe as each shell file append calls an (open, write, close) triplet.

Reply:
> I haven't checked the source, but strace suggests this is safe as each shell file append calls an (open, write, close) triplet.

It would be safe only if (open, write, close) are done atomically.